### DISCUSSION TITLE
opp-no3-Discussions!_baitapgiuaki

### DISCUSSION THREAD

**Discussion by deptrainhatgai** [deptrainhatgai/NHOM14_N03#3] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by deptrainhatgai** [comment]
..

**Comment by nghongthai** [comment]
public interface Signal {
    void activate();
    void deactivate();
}
public class Main {
    public static void main(String[] args) {
        Signal discrete = new DiscreteSignal();
        Signal continuous = new ContinuousSignal();
        
        discrete.activate();
        discrete.deactivate();
        
        continuous.activate();
        continuous.deactivate();
    }
}
public class ContinuousSignal implements Signal {
    @Override
    public void activate() {
        System.out.println("ContinuousSignal is activated");
    }

    @Override
    public void deactivate() {
        System.out.println("ContinuousSignal is deactivated");
    }
}
public class DiscreteSignal implements Signal {
    @Override
    public void activate() {
        System.out.println("DiscreteSignal is activated");
    }

    @Override
    public void deactivate() {
        System.out.println("DiscreteSignal is deactivated");
    }
}

**Comment by deptrainhatgai** [comment]
phải tách ra chứ

**Comment by deptrainhatgai** [comment]
bị lỗi rồi

**Reply by nglthu** [reply]
Error message ntn ? @deptrainhatgai

**Comment by deptrainhatgai** [comment]
sửa @overRide thành @override

**Comment by deptrainhatgai** [comment]
public interface Signal {
    void activate();
    void deactivate();
    }

**Comment by deptrainhatgai** [comment]
public class DiscreteSignal implements Signal {
    @Override
    public void activate() {
    System.out.println("DiscreteSignal is activated");
    }
    
    @Override
    public void deactivate() {
        System.out.println("DiscreteSignal is deactivated");
    }
    }

**Comment by deptrainhatgai** [comment]
public class ContinuousSignal implements Signal {
    @Override
    public void activate() {
    System.out.println("ContinuousSignal is activated");
    }
    
    @Override
    public void deactivate() {
        System.out.println("ContinuousSignal is deactivated");
    }
    }

**Comment by deptrainhatgai** [comment]
public class Main {
    public static void main(String[] args) {
        DiscreteSignal discreteSignal = new DiscreteSignal();

        discreteSignal.activate();

        discreteSignal.deactivate();

        ContinuousSignal continuousSignal = new ContinuousSignal();

        continuousSignal.activate(); 

        continuousSignal.deactivate();  
    }
}

**Comment by Sangisme108** [comment]
mac@MacBook-Pro-cua-Mac gk oop.java % cd "/Users/mac/Documents/gk oop.java/" && javac ContinuousSignal.java && java ContinuousSignal
ContinuousSignal.java:2: error: cannot find symbol
    @OverRide
     ^
  symbol:   class OverRide
  location: class ContinuousSignal
1 error
mac@MacBook-Pro-cua-Mac gk oop.java %  lỗi override

**Reply by nglthu** [reply]
bỏ @override di

**Reply by Sangisme108** [reply]
dạ vg đc r ạ

**Comment by nghongthai** [comment]
public class Radar {
    // Phương thức phân tích tín hiệu với mẫu tín hiệu
    public double[] analyzeSignal(int size) {
        double[] signal = new double[size];
        for (int n = 0; n < size; n++) {
            if (n >= 0 && n <= 15) {
                signal[n] = 1 - (n / 15.0);
            } else {
                signal[n] = 0;
            }
        }
        return signal;
    }

    public static void main(String[] args) {
        Radar radar = new Radar();
        double[] analyzedSignal = radar.analyzeSignal(4); // Phân tích tín hiệu với kích thước n = 4
        System.out.println("Radar Analyzed Signal:");
        for (double value : analyzedSignal) {
            System.out.print(value + " ");
        }
        System.out.println();
    }
}

**Comment by nghongthai** [comment]
ý 1;
public class DiscreteSignal implements Signal {
    private int k; // Độ trễ (delay)

    // Constructor để khởi tạo độ trễ k
    public DiscreteSignal(int k) {
        this.k = k;
    }

    // Phương thức tính giá trị của x(n) dựa vào x(n - k)
    public int compute(int n, int[] x) {
        int d_n = (n == 0) ? 1 : 0; // Dãy xung đơn vị d(n)
        if (n - k >= 0) {
            return d_n * x[n - k];
        }
        return 0;
    }

    
    public int[] generateSignal(int size) {
        int[] signal = new int[size];
        for (int i = 0; i < size; i++) {
            signal[i] = compute(i, signal);
        }
        return signal;
    }

    
    public void activate() {
        System.out.println("DiscreteSignal is activated");
    }

    
    public void deactivate() {
        System.out.println("DiscreteSignal is deactivated");
    }
}

**Comment by nghongthai** [comment]
public class Main {
    public static void main(String[] args) {
        // Tạo và thử nghiệm DiscreteSignal
        DiscreteSignal discreteSignal = new DiscreteSignal(2); // Ví dụ k = 2
        int[] signal = discreteSignal.generateSignal(10);
        System.out.println("Discrete Signal:");
        for (int value : signal) {
            System.out.print(value + " ");
        }
        System.out.println();
        
        discreteSignal.activate();
        discreteSignal.deactivate();

        // Tạo và thử nghiệm ContinuousSignal
        ContinuousSignal continuousSignal = new ContinuousSignal();
        continuousSignal.activate();
        continuousSignal.deactivate();
        
        // Thử nghiệm Radar
        Radar radar = new Radar();
        double[] analyzedSignal = radar.analyzeSignal(4); // Phân tích tín hiệu với kích thước n = 4
        System.out.println("Radar Analyzed Signal:");
        for (double value : analyzedSignal) {
            System.out.print(value + " ");
        }
        System.out.println();
    }
}

**Comment by nghongthai** [comment]
public class DiscreteSignal implements Signal {
    private int k; // Độ trễ (delay)

    // Constructor để khởi tạo độ trễ k
    public DiscreteSignal(int k) {
        this.k = k;
    }

    // Phương thức tính giá trị của x(n) dựa vào x(n - k)
    public int compute(int n, int[] x) {
        int d_n = (n == 0) ? 1 : 0; // Dãy xung đơn vị d(n)
        if (n - k >= 0) {
            return d_n * x[n - k];
        }
        return 0;
    }

    
    public int[] generateSignal(int size) {
        int[] signal = new int[size];
        for (int i = 0; i < size; i++) {
            signal[i] = compute(i, signal);
        }
        return signal;
    }

    
    public void activate() {
        System.out.println("DiscreteSignal is activated");
    }

    
    public void deactivate() {
        System.out.println("DiscreteSignal is deactivated");
    }
}

**Comment by nghongthai** [comment]
public class ContinuousSignal implements Signal {
   
    public void activate() {
        System.out.println("ContinuousSignal is activated");
    }

  
    public void deactivate() {
        System.out.println("ContinuousSignal is deactivated");
    }
}

**Comment by nghongthai** [comment]
public class Main {
    public static void main(String[] args) {
        // Tạo và thử nghiệm DiscreteSignal
        DiscreteSignal discreteSignal = new DiscreteSignal(2); // Ví dụ k = 4
        int[] signal = discreteSignal.generateSignal(10);
        System.out.println("Discrete Signal:");
        for (int value : signal) {
            System.out.print(value + " ");
        }
        System.out.println();
        
        discreteSignal.activate();
        discreteSignal.deactivate();

        // Tạo và thử nghiệm ContinuousSignal
        // ContinuousSignal continuousSignal = new ContinuousSignal();
        // continuousSignal.activate();
        // continuousSignal.deactivate();
        
        // Thử nghiệm Radar
        Radar radar = new Radar();
        double[] analyzedSignal = radar.analyzeSignal(4); // Phân tích tín hiệu với kích thước n = 4
        System.out.println("Radar Analyzed Signal:");
        for (double value : analyzedSignal) {
            System.out.print(value + " ");
        }
        System.out.println();
    }
}
